Replace `constraints` with `dynamic_shapes` in test/inductor

BoyuanFeng opened the pull request:
Summary:
`constraints` argument for `torch.export` has been deprecated in favor of the `dynamic_shapes` argument. This PR updates the use of the deprecated API in `scripts/sijiac/prototypes`.

Test Plan: buck test mode/dev-nosan fbcode//caffe2/test/inductor:test_aot_inductor

Differential Revision: D52790713




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

avikchaudhuri reviewed (approved):
Fix another failing test

BoyuanFeng committed:
Replace `constraints` with `dynamic_shapes` in scripts/sijiac/prototypes and test/inductor (#117779)

Summary:

`constraints` argument for `torch.export` has been deprecated in favor of the `dynamic_shapes` argument. This PR updates the use of the deprecated API in `scripts/sijiac/prototypes` and `test/inductor`.

Test Plan: buck test mode/dev-nosan fbcode//caffe2/test/inductor:test_aot_inductor

Reviewed By: avikchaudhuri, angelayi

Differential Revision: D52790713
